tests/wasmfs/wasmfs_open.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+// FIXME: Merge with other existing close and open tests.
+
+int main() {
+  // Test writing to a file with a trailing slash.
+  int fd = open("/dev/stdout/", O_RDONLY);
+
+  dprintf(fd, "WORKING WITH TRAILING BACKSLASH\n");
+
+  // Close open file
+  close(fd);
+
+  // Test writing to a file with no trailing backslash.
+  int fd2 = open("/dev/stdout", O_RDONLY);
+
+  dprintf(fd2, "WORKING WITHOUT TRAILING BACKSLASH\n");
+
+  // Close open file
+  close(fd2);
+
+  // Attempt to write to a non-existent fd.
+  errno = 0;
+  dprintf(fd, "SHOULD NOT PRINT\n");
+  assert(errno == EBADF);
+  printf("Errno: %s\n", strerror(errno));
+
+  // Attempt to open and then read/write to a directory.
+  int fd3 = open("/dev", O_RDONLY);
+
+  const char* msg = "Test\n";
+
+  errno = 0;
+  write(fd3, msg, strlen(msg));
+  assert(errno == EISDIR);
```

Comment:
Testing locally, it looks like this returns EBADF because the file was opened for reading only rather than EISDIR.

Reply:
Yeah I can change this to `O_RDWR`

Reply:
I think it would be better to update the test expectation; IIRC, opening a directory with O_RDWR is supposed to fail so you wouldn't even be able to call write on it.

Reply:
Oh that's right.